backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java

```diff
@@ -484,9 +484,11 @@ public Response updateEndpoint(@Context SecurityContext sec,
                 Log.error("RHOSE processor update failed", e);
                 return Response.serverError().build();
             }
-        }
 
-        endpointRepository.updateEndpoint(endpoint);
+            endpointRepository.updateEndpoint(dbEndpoint);
+        } else {
+            endpointRepository.updateEndpoint(endpoint);
```

Comment:
This isn't pretty but at least this code does what we need. We may refactor the entire endpoint update logic and replace it with something cleaner later...